x/ccv/consumer/keeper/keeper.go

```diff
@@ -206,50 +206,95 @@ func (k Keeper) DeletePendingChanges(ctx sdk.Context) {
 	store.Delete(types.PendingChangesKey())
 }
 
-// IteratePacketMaturityTime iterates through the VSC packet maturity times set in the store
-func (k Keeper) IteratePacketMaturityTime(ctx sdk.Context, cb func(vscId, timeNs uint64) (stop bool)) {
+// GetVSCPacketQueueTimeSlice gets a specific VSC packet queue timeslice.
+// A timeslice is a slice of IDs corresponding to received VSC packets
+// that reach maturity at a certain time.
+func (k Keeper) GetVSCPacketQueueTimeSlice(ctx sdk.Context, timestamp time.Time) (vscIDs []uint64) {
 	store := ctx.KVStore(k.storeKey)
-	iterator := sdk.KVStorePrefixIterator(store, []byte{types.PacketMaturityTimeBytePrefix})
 
-	defer iterator.Close()
-	for ; iterator.Valid(); iterator.Next() {
-		// Extract bytes following the 1 byte prefix
-		seqBytes := iterator.Key()[1:]
-		seq := binary.BigEndian.Uint64(seqBytes)
+	bz := store.Get(types.VSCPacketQueueKey(timestamp))
+	if bz == nil {
+		return []uint64{}
+	}
 
-		timeNs := binary.BigEndian.Uint64(iterator.Value())
+	validatorSetChangeIDs := types.ValidatorSetChangeIDs{}
+	k.cdc.MustUnmarshal(bz, &validatorSetChangeIDs)
 
-		stop := cb(seq, timeNs)
-		if stop {
-			break
-		}
+	return validatorSetChangeIDs.Ids
+}
+
+// SetVSCPacketQueueTimeSlice sets a specific VSC packet queue timeslice.
+func (k Keeper) SetVSCPacketQueueTimeSlice(ctx sdk.Context, timestamp time.Time, vscIDs ...uint64) {
+	store := ctx.KVStore(k.storeKey)
+	bz := k.cdc.MustMarshal(&types.ValidatorSetChangeIDs{Ids: vscIDs})
+	store.Set(types.VSCPacketQueueKey(timestamp), bz)
+}
+
+// InsertVSCPacketQueue inserts the ID of a VSC packet to the appropriate timeslice
+// in the VSCPacketQueue.
+func (k Keeper) InsertVSCPacketQueue(ctx sdk.Context, vscID uint64, maturityTime time.Time) {
```

Comment:
Continuing on comment above, if we need both functions, can this be refactored to:
```golang
func (k Keeper) InsertVSCPacketQueue(ctx sdk.Context, vscID uint64, maturityTime time.Time) {
	timeSlice := append(k.GetVSCPacketQueueTimeSlice(ctx, maturityTime), vscID)
	k.SetVSCPacketQueueTimeSlice(ctx, maturityTime, timeSlice...)
}
```

Reply:
I removed `SetVSCPacketQueueTimeSlice`